nf_core/pipeline-template/{{cookiecutter.name_noslash}}/lib/JSON.groovy

```diff
@@ -0,0 +1,117 @@
+/*
+ * This file holds several functions used to perform JSON parameter validation, help and summary rendering for the nf-core pipeline template.
+ */
+
+import groovy.json.JsonSlurper
+
+class JSON {
+    /*
+    * This method tries to read a JSON params file
+    */
+    private static LinkedHashMap params_get(String path) {
+        def usage = new LinkedHashMap()
+        try {
+            usage = params_try(path)
+        } catch (Exception e) {
+            println "Could not read parameters settings from JSON. $e"
+            usage = new LinkedHashMap()
+        }
+        return usage
+    }
+
+    /*
+    Method to actually read in JSON file using Groovy.
+    Group (as Key), values are all parameters
+        - Parameter1 as Key, Description as Value
+        - Parameter2 as Key, Description as Value
+        ....
+    Group
+        -
+    */
+    private static LinkedHashMap params_try(String path) throws Exception {
+
+        def json = new File(path).text
+        def Map usage = (Map) new JsonSlurper().parseText(json).get('properties')
+
+        /* Tree looks like this in nf-core schema
+        *  properties <- this is what the first get('properties') gets us
+             group 1
+               properties
+               description
+             group 2
+               properties
+               description
+             group 3
+               properties
+               description
+        */
+        def output_map = new LinkedHashMap()
+
+        //Lets go deeper
+        usage.each { key, val ->
+            //println "Group: $key\n" //Gets the Group names
+            def Map submap = usage."$key".properties //Gets the property object of the group
+            //println "Parameters:\n" //nested access to the Parameters
+            def sub_params = new LinkedHashMap()
+            submap.each { innerkey, value ->
+                //println "$innerkey\t$value.description"
+                sub_params.put("$innerkey", "$value.description")
+            }
+            output_map.put("$key", sub_params)
+            //println "\n"
+        }
+        return output_map
+    }
+
+    static String params_help(path, command) {
+          String output = "Typical pipeline command:\n\n"
+          output += "    ${command}\n\n"
+          output += params_beautify(params_get(path))
+    }
+
+    static String params_beautify(usage) {
+        String output = ""
+        for (group in usage.keySet()) {
+            output += group + "\n"
+            def params = usage.get(group) //This gets the parameters of that particular group
+            for (par in params.keySet()) {
+                output+= "    \u001B[1m" + par.padRight(27) + "\u001B[1m" + params.get(par) + "\n"
```

Comment:
Can we auto-detect the maximum parameter length instead of hardcoding `27`?

Reply:
So alot of the JSON stuff I left alone because you said you wanted to play with that 😅 Thats why I havent deleted comments and such just in case they are handy for you.

Reply:
Ok cool ✨